swatch-system-conduit/src/main/java/org/candlepin/subscriptions/conduit/InventoryController.java

```diff
@@ -280,6 +282,7 @@ private void extractNetworkFacts(Map<String, String> rhsmFacts, ConduitFacts fac
       facts.setFqdn(fqdn);
     }
 
+    facts.setNetworkInterfaces(populateNICs(rhsmFacts));
```

Comment:
We should not set `networkInterfaces` if `populateNics(rhsmFacts)` returns an empty list.
```suggestion
    List<NetworkInterface> networkInterfaces = populateNICs(rhsmFacts);
    if (!networkInterfaces.isEmpty()) {
      facts.setNetworkInterfaces(networkInterfaces);
    }
```

This will address the test failures you were seeing with comparing `networkInterfaces` in `equals`.